src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -13,6 +13,12 @@ public class StringFunctions {
     private static final char SPACE = ' ';
 
     public static final String reverseWithStringConcat(String string) {
+
+        // Check if the input string is null or empty, and return an empty string if true
+        if (string == null || string.isEmpty()) {
```

Comment:
I added this if block so that if the string is null or if the string is empty the program can handle these cases without crashing or outputting an unexpected output.